File: leetcode/346.moving-average-from-data-stream.cpp
class MovingAverage {
private:
    deque<int> dq;
    int size;
    int sum;
public:
    /** Initialize your data structure here. */
    MovingAverage(int size) {
        this->size = size;
        this->sum = 0;
    }
    
    double next(int val) {
        
        dq.push_back(val);
        sum += val;
        
        if (dq.size() > size)
        {
            sum -= dq.front();
            dq.pop_front();
        }
        
        return double(sum) / dq.size();
    }
};

/**
 * Your MovingAverage object will be instantiated and called as such:
 * MovingAverage obj = new MovingAverage(size);
 * double param_1 = obj.next(val);
 */

File: leetcode/346.moving-average-from-data-stream.py
from collections import deque
class MovingAverage(object):

    def __init__(self, size):
        """
        Initialize your data structure here.
        :type size: int
        """
        self.dq = deque()
        self.size = size
        self.sum = 0

    def next(self, val):
        """
        :type val: int
        :rtype: float
        """
        self.dq.append(val)
        self.sum += val
        if len(self.dq) > self.size:
            self.sum -= self.dq.popleft()
            
        return float(self.sum) / len(self.dq)


# Your MovingAverage object will be instantiated and called as such:
# obj = MovingAverage(size)
# param_1 = obj.next(val)